lib/rex/socket.rb

```diff
@@ -963,6 +993,39 @@ def self.rex_resolve_hostname(name, resolver: @@resolver)
     # Ensure response types (depending on underlying library used) provide required methods
     return v4, v6
   end
+
+  def self.rex_getresources(name, typeclass, resolver: @@resolver)
+    raise ::SocketError.new(
+      "Rex::Socket internal DNS resolution requires passing/setting a resolver"
+    ) unless resolver
+    raise ::SocketError.new(
+      "Rex::Socket internal DNS resolution requires passing a String name to resolve"
+    ) unless name.is_a?(String)
+
+    typeclass = typeclass.upcase
+    attribute = {
+      CNAME: :domainname,
+      MX:    :exchange,
+      NS:    :domainname,
+      PTR:   :domainname,
+      SOA:   :mname,
+      SRV:   :target
+    }[typeclass]
+    if attribute.nil?
+      raise ArgumentError, 'Invalid typeclass'
+    end
+    const = ::Net::DNS.const_get(typeclass)
+
+    resources = begin
+      resolver.send(name, const).answer.select do |a|
+        a.type == const
```

Comment:
Is this `resources` object symmetrical with the one above, or just has fitting accessors for interop? Structural mismatches between the two can manifest in weird ways down the line, so if they do diverge probably a good idea to annotate in a comment somewhere.

Reply:
In both cases, the return value is an array of strings.

Reply:
Thanks boss. I say ship it - not a breaking change, new feature so I figure we can iron it out on real use cases as we go.